tests/foreman/cli/test_subnet.py

```diff
@@ -69,532 +72,537 @@ def invalid_missing_attributes():
     ]
 
 
-class SubnetTestCase(CLITestCase):
-    """Subnet CLI tests.
+@pytest.mark.tier1
+@pytest.mark.upgrade
+def test_positive_CRUD():
+    """Create, update and delete subnet
+
+    :id: d74a52a7-df56-44ef-89a3-081c14e81e43
+
+    :expectedresults: Subnet is created, updated and deleted
+
+    :CaseImportance: Critical
+    """
+    name = gen_choice(list(valid_data_list().values()))
+    pool = sorted(valid_addr_pools()[0])
+    mask = '255.255.255.0'
+    # generate pool range from network address
+    network = gen_ipaddr()
+    from_ip = re.sub(r'\d+$', str(pool[0]), network)
+    to_ip = re.sub(r'\d+$', str(pool[1]), network)
+    domains_amount = random.randint(3, 5)
+    domains = [make_domain() for _ in range(domains_amount)]
+    gateway = gen_ipaddr(ip3=True)
+    ipam_type = SUBNET_IPAM_TYPES['dhcp']
+    subnet = make_subnet(
+        {
+            'name': name,
+            'from': from_ip,
+            'mask': mask,
+            'network': network,
+            'to': to_ip,
+            'domain-ids': [domain['id'] for domain in domains],
+            'gateway': gateway,
+            'ipam': ipam_type,
+        }
+    )
+    # Check if Subnet can be listed
+    subnets_ids = [subnet_['id'] for subnet_ in Subnet.list()]
+    assert subnet['id'] in subnets_ids
+    assert subnet['name'] == name
+    assert subnet['start-of-ip-range'] == from_ip
+    assert subnet['end-of-ip-range'] == to_ip
+    assert len(subnet['domains']) == domains_amount
+    for domain in domains:
+        assert domain['name'] in subnet['domains']
+    assert gateway in subnet['gateway-addr']
+    assert ipam_type in subnet['ipam']
+
+    # update subnet
+    new_name = gen_choice(list(valid_data_list().values()))
+    pool = sorted(valid_addr_pools()[0])
+    # generate pool range from network address
+    new_network = gen_ipaddr()
+    new_mask = '255.255.192.0'
+    ip_from = re.sub(r'\d+$', str(pool[0]), new_network)
+    ip_to = re.sub(r'\d+$', str(pool[1]), new_network)
+    ipam_type = SUBNET_IPAM_TYPES['internal']
+    Subnet.update(
+        {
+            'new-name': new_name,
+            'from': ip_from,
+            'id': subnet['id'],
+            'to': ip_to,
+            'mask': new_mask,
+            'network': new_network,
+            'ipam': ipam_type,
+            'domain-ids': "",  # delete domains needed for subnet delete
+        }
+    )
+    subnet = Subnet.info({'id': subnet['id']})
+    assert subnet['name'] == new_name
+    assert subnet['start-of-ip-range'] == ip_from
+    assert subnet['end-of-ip-range'] == ip_to
+    assert subnet['network-addr'] == new_network
+    assert subnet['network-mask'] == new_mask
+    assert ipam_type in subnet['ipam']
+
+    # delete subnet
+    Subnet.delete({'id': subnet['id']})
+    with pytest.raises(CLIReturnCodeError):
+        Subnet.info({'id': subnet['id']})
+
+
+@pytest.mark.tier2
+@pytest.mark.parametrize('options', **parametrized(invalid_missing_attributes()))
```

Comment:
Make sure parameterized tests have their docstrings updated to indicate they are.